### ISSUE TITLE
react-dom typing errors

### ISSUE THREAD

**Issue by trodrigues** [DefinitelyTyped/DefinitelyTyped#17800]
- [x] I tried using the `@types/react-dom` package and had problems.
- [x] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript (2.4.1)
- [x] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).
- [ ] [Mention](https://github.com/blog/821-mention-somebody-they-re-notified) the authors (see `Definitions by:` in `index.d.ts`) so they can respond.
  - Authors: @MartynasZilinskas @tkrotoff @andy-ms 

```
node_modules/@types/react-dom/index.d.ts(42,22): error TS2314: Generic type 'Component<P, S>' requires 2 type argument(s).
node_modules/@types/react-dom/index.d.ts(57,22): error TS2314: Generic type 'Component<P, S>' requires 2 type argument(s).
node_modules/@types/react-dom/index.d.ts(62,22): error TS2314: Generic type 'Component<P, S>' requires 2 type argument(s).
node_modules/@types/react-dom/index.d.ts(67,22): error TS2314: Generic type 'Component<P, S>' requires 2 type argument(s).
```

**Comment by trodrigues** [comment]
Also to add that this was clearly introduced in 15.5.0 as 15.5.1 does not have this issue.

react-dom version is 15.6.1.

**Comment by tkrotoff** [comment]
Simply update your react types package to the latest version

**Comment by trodrigues** [comment]
I had tried that before but I'm also stuck on an older version of react-router which has some issues with that latest version of the typings. However even with that version I was still seeing these react-dom warnings there.

I'll report back when I've managed to update all of that as given that the react-router v3 to v4 update doesn't have such a straightforward migration path this might be a bit more complicated.

**Comment by Ciantic** [comment]
I don't think this is so simple, I'm getting exact same errors and I have this:

```
    "name": "@types/react",
    "version": "15.0.35",
```

and I have `"@types/react": "^15.0.35",` in package json.

Which is the latest version.

**Comment by Ciantic** [comment]
there is something funny going on in here, maybe this is related to yarn. Did the parent @trodrigues  also used yarn?

**Comment by gsdnano** [comment]
Having same issue. Using NPM. Latest versions on everythiing (fresh install).

**Comment by mayerwin** [comment]
Same issue here. With latest versions on everything...

**Comment by andy-ms** [comment]
@mayerwin Could you look at the definition for `Component` that's giving you the error? The latest definition is `interface Component<P = {}, S = {}> extends ComponentLifecycle<P, S> { }`, which shouldn't require all type arguments to be supplied (assuming you are using at least typescript@2.3).

**Comment by mayerwin** [comment]
@andy-ms, it is indeed this definition, I solved the issue by updating Resharper.

**Comment by orta** [comment]
Hi thread, we're moving DefinitelyTyped to use [GitHub Discussions](https://github.com/DefinitelyTyped/DefinitelyTyped/issues/53377) for conversations the `@types` modules in DefinitelyTyped.

To help with the transition, we're closing all issues which haven't had activity in the last 6 months, which includes this issue. If you think closing this issue is a mistake, please pop into the [TypeScript Community Discord](https://discord.gg/typescript) and mention the issue in the `definitely-typed` channel.
